sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -76,10 +77,21 @@ static void _sleep_radio(gnrc_lorawan_t *mac)
     dev->driver->set(dev, NETOPT_STATE, &state, sizeof(state));
 }
 
-void gnrc_lorawan_init(gnrc_lorawan_t *mac, uint8_t *nwkskey, uint8_t *appskey)
+void gnrc_lorawan_init(gnrc_lorawan_t *mac, const gnrc_lorawan_ctx_t *ctx)
 {
-    mac->nwkskey = nwkskey;
-    mac->appskey = appskey;
+    mac->joineui = ctx->joineui;
+    mac->appskey = ctx->appskey;
+
+#if IS_USED(MODULE_GNRC_LORAWAN_1_1)
+    mac->fnwksintkey = ctx->fnwksintkey;
+    mac->snwksintkey = ctx->snwksintkey;
+    mac->nwksenckey = ctx->nwksenckey;
+    mac->jsintkey = ctx->jsintkey;
+    mac->jsenckey = ctx->jsenckey;
+#else
+    mac->fnwksintkey = mac->snwksintkey = mac->nwksenckey = ctx->fnwksintkey;
+#endif
+
```

Comment:
It seems all members of `gnrc_lorawan_ctx_t` end up in the MAC descriptor. Therefore, have you considered storing the pointer there? Then you can simply do:
```suggestion
    memcpy(&mac->ctx, ctx, sizeof(gnrc_lorawan_ctx_t));
```

That way you remove the #if macro here. Of course you will need to set all `ctx` members to `ctx->fnwksintkey`, but you can do that before calling `gnrc_lorawan_init`